nixos/modules/services/networking/pdns-recursor.nix

```diff
@@ -117,17 +119,53 @@ in {
       '';
     };
 
-    extraConfig = mkOption {
+    settings = mkOption {
+      type = configType;
+      default = { };
+      example = literalExample ''
+        {
+          lua-config-file = pkgs.writeText "recursor.lua" '''
+            -- lua code here
+          ''';
+        }
+      '';
+      description = ''
+        PowerDNS recursor settings. Use this option to configure recursor
+        settings not exposed in a NixOS option or to bypass one.
+        See the full documentation at
+        <link xlink:href="https://doc.powerdns.com/recursor/settings.html"/>
+        for the available options.
+      '';
+    };
+
+    luaConfig = mkOption {
       type = types.lines;
       default = "";
       description = ''
-        Extra options to be appended to the configuration file.
+        The content Lua configuration file for PowerDNS recursor. See
+        <link xlink:href="https://doc.powerdns.com/recursor/lua-config/index.html"/>.
       '';
     };
   };
 
   config = mkIf cfg.enable {
 
+    services.pdns-recursor.settings = mkDefaultAttrs {
```

Comment:
You probably shouldn't use `mkDefault` for `lua-config-file`, because the user shouldn't this easily be able to override the text in `luaConfig`. I'd also suggest the `settings` option to have a different example because of that

Reply:
I get that but it makes more sense for all options to have the same priority... This could also apply to `forward-zones` since it's values is generated by Nix.

Reply:
Hm, I just worry that customizations done to `luaConfig` by other modules like https://github.com/NixOS/nixpkgs/pull/58204 get cleared too easily. But this might be alright

Reply:
You're right, but at the same time It won't be great to have users need an `mkForce` to change settings and raising the priority of only some options seems arbitrary. I also thought about a warning like "you are directly using `settings.lua-config` do you know of `luaConfig`?" but this is probably an even greater annoyance: it will comes up on every build.
I'd like to keep it this way if it's ok for you.

Reply:
Alright sounds good

Reply:
Thank you.